homeassistant/components/bme280/sensor.py

```diff
@@ -146,6 +152,15 @@ def __init__(self, bme280_client, sensor_type, temp_unit, name):
         self.type = sensor_type
         self._state = None
         self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+        super().__init__(
+            DataUpdateCoordinator(
+                hass,
+                _LOGGER,
+                name=DOMAIN,
+                update_method=self.async_update,
```

Comment:
The update method should no be an entity method. The coordinator will tell the entity to update when the coordinator has new state. The entity should not poll.